presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PlanNodeDecorrelator.java

```diff
@@ -145,8 +145,12 @@ public PlanNodeDecorrelator(PlanNodeIdAllocator idAllocator, Lookup lookup)
         @Override
         public Optional<DecorrelationResult> visitLimit(LimitNode node, Void context)
         {
+            if (node.getCount() == 0) {
+                return Optional.empty();
```

Comment:
actually such subquery should be easy to be decorrelated.

Isn't this a deadcode? Maybe we should replace that with `verify`?

Reply:
I just opportunistically moved the condition so we check it early. Anything beyond that is beyond scope.